sources/imperative/reader/ReaderAPI.swift

```diff
@@ -49,6 +49,21 @@ extension CSVReader {
         } else {
             // B. Otherwise, create an input stream and start parsing byte-by-byte.
             guard let stream = InputStream(url: input) else { throw Error._invalidFile(url: input) }
+            try self.init(input: stream, configuration: configuration)
+        }
+    }
+
+    /// Creates a reader instance that will be used to parse the given CSV stream.
+    ///
+    /// If the configuration's encoding hasn't been set and the input data doesn't contain a Byte Order Marker (BOM), UTF8 is presumed.
+    /// - parameter input: The stream to be parsed.
+    /// - parameter configuration: Recipe detailing how to parse the CSV data (i.e. encoding, delimiters, etc.).
+    /// - throws: `CSVError<CSVReader>` exclusively.
+    public convenience init(input stream: InputStream, configuration: Configuration = .init()) throws {
```

Comment:
Alright, reverted my change back to what I originally had. The public interface for `InputStream` is back.